### PR TITLE
Install Postgres 9.4

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**

### CODE REVIEW

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco please take a look... theres a lot of crap thats been commented out, but please ignore this.

**Comment by michaelachrisco** [comment]
Any reason for not using https://github.com/phlipper/chef-postgresql to install PostgreSQL?

**Comment by michaelachrisco** [comment]
I am also assuming this is Ubuntu 12.04 since you are using the default `postgresql-common` and not the one for 13.04?

File: site-cookbooks/postgresql/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#   libxml2-dev
+#   libxslt-dev}
+
+# This works, but I'm not sure exactly why... which packages below are needed
+# for postgresql-9-4 to install?
+packages = %w{
+  libpq5
+  pgdg-keyring
+  postgresql-client-common
+  postgresql-9.4}
```
Comment (michaelachrisco) [inline site-cookbooks/postgresql/recipes/default.rb:28]: Some of the guides recommend `libpq-dev` as well but have no reason why (that I have seen). Ill look into it.

**Comment by josephbridgwaterrowe** [comment]
Ubuntu 14.04

**Comment by josephbridgwaterrowe** [comment]
To be honest it was a bit of a punt since what should of been trivial turned out to be not so trivial. At the time I couldn't find any good documentation.

**Comment by michaelachrisco** [comment]
Ok cool. 14.04 has it under apt-get :+1:.

**Comment by josephbridgwaterrowe** [comment]
The reason I've avoiding any other existing cookbooks is because they typically rely on json databags for configuration. Since we're not running a server this could be a pain to manage since I don't want these things in source control.

**Comment by josephbridgwaterrowe** [comment]
14.04 has 9.4 now?

**Comment by josephbridgwaterrowe** [comment]
I had to add in the postgresql repo at the time to install 9.4 :sad:

**Comment by michaelachrisco** [comment]
`apt-get install postgresql-9.4`

**Comment by michaelachrisco** [comment]
Still, using the official repo is probably better for updates.

**Comment by josephbridgwaterrowe** [comment]
True, but the Ubuntu repo would probably be easier :smile:

File: site-cookbooks/postgresql/recipes/default.rb (hunk 1 above)
Comment (michaelachrisco) [inline site-cookbooks/postgresql/recipes/default.rb:28]: Ok just come C headers that we don't need. Disregard.

**Comment by michaelachrisco** [comment]
Indeed. And possibly works better on Ubuntu:question:

**Comment by josephbridgwaterrowe** [comment]
I don't want to make changes to this without testing, I don't really have an environment for this at the moment.

Further down the road when we have a Chef server I see us using TestKitchen to test our cookbooks, but we'll probably use alot of off the shelf cookbooks when we move to a server.

**Comment by josephbridgwaterrowe** [comment]
Perhaps, although the postgresql package is for the trusty distro so hopefully its been configured for that distro?

**Comment by michaelachrisco** [comment]
Personally, if it works, then I am alright using your chef ATM. We can change it later if there is some recommended way of installing PG that gives us rainbows and butterflies. 

Besides the branch is pretty simple and would not require a whole lot to rip out.

**Comment by michaelachrisco** [comment]
This looks good for merging. Agreed?

**Comment by josephbridgwaterrowe** [comment]
Yes, well its shit, but it works.

**Pull request merged**